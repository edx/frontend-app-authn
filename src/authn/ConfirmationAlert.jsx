import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from '@edx/frontend-platform/i18n';
import { Alert, Hyperlink } from '@edx/paragon';

const ConfirmationAlert = (props) => {
  const { email } = props;

  const technicalSupportLink = (
    <Hyperlink
      destination="https://support.edx.org/hc/en-us/articles/206212088-What-if-I-did-not-receive-a-password-reset-message-"
    >
      <FormattedMessage
        id="authn.forgot.password.confirmation.support.link"
        defaultMessage="contact technical support"
        description="link text used in message: authn.forgot.password.confirmation.support.link 'contact technical support.'"
      />
    </Hyperlink>
  );

  const strongEmail = (<strong>{email}</strong>);
  const lineBreak = (<br />);

  return (
    <Alert
      variant="success"
    >
      <Alert.Heading className="text-success">
        <FormattedMessage
          id="authn.forgot.password.confirmation.title"
          defaultMessage="Check Your Email"
          description="Forgot password confirmation title"
        />
      </Alert.Heading>
      <FormattedMessage
        id="authn.forgot.password.confirmation.message"
        defaultMessage="You entered {strongEmail}. If this email address is associated with your edX account, we will send a message with password recovery instructions to this email address. {lineBreak}If you do not receive a password reset message after 1 minute, verify that you entered the correct email address, or check your spam folder.{lineBreak} If you need further assistance, {technicalSupportLink}."
        description="Forgot password confirmation message"
        values={{
          strongEmail,
          technicalSupportLink,
          lineBreak,
        }}
      />
    </Alert>
  );
};

ConfirmationAlert.propTypes = {
  email: PropTypes.string.isRequired,
};

export default ConfirmationAlert;
