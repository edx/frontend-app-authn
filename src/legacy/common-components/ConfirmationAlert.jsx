import React from 'react';
import PropTypes from 'prop-types';

import { getConfig } from '@edx/frontend-platform';
import { FormattedMessage, injectIntl, intlShape } from '@edx/frontend-platform/i18n';
import { Alert } from '@edx/paragon';

import messages from './messages';

const ConfirmationAlert = (props) => {
  const { email, intl } = props;

  return (
    <Alert id="confirmation-alert" variant="success">
      <Alert.Heading>{intl.formatMessage(messages['forgot.password.confirmation.title'])}</Alert.Heading>
      <p>
        <FormattedMessage
          id="forgot.password.confirmation.message"
          defaultMessage="You entered {strongEmail}. If this email address is associated with your
          edX account, we will send a message with password recovery instructions to this email address."
          description="Forgot password confirmation message"
          values={{ strongEmail: <strong className="data-hj-suppress">{email}</strong> }}
        />
      </p>
      <p>{intl.formatMessage(messages['forgot.password.confirmation.info'])}</p>
      <p>
        <FormattedMessage
          id="forgot.password.technical.support.help.message"
          defaultMessage="If you need further assistance, {technicalSupportLink}."
          description="Message to help user contact technical support"
          values={{
            technicalSupportLink: (
              <Alert.Link href={getConfig().PASSWORD_RESET_SUPPORT_LINK}>
                {intl.formatMessage(messages['forgot.password.confirmation.support.link'])}
              </Alert.Link>
            ),
          }}
        />
      </p>
    </Alert>
  );
};

ConfirmationAlert.propTypes = {
  email: PropTypes.string.isRequired,
  intl: intlShape.isRequired,
};

export default injectIntl(ConfirmationAlert);
