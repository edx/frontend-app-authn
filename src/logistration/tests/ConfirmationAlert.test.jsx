import React from 'react';
import { mount } from 'enzyme';

import { mergeConfig } from '@edx/frontend-platform';
import { injectIntl, IntlProvider } from '@edx/frontend-platform/i18n';

import ConfirmationAlert from '../ConfirmationAlert';

const IntlConfirmationAlertMessage = injectIntl(ConfirmationAlert);

describe('ConfirmationAlert', () => {
  const supportLink = 'https://support.test.com/What-if-I-did-not-receive-a-password-reset-message';
  mergeConfig({
    PASSWORD_RESET_SUPPORT_LINK: supportLink,
  });

  it('should match default confirmation message', () => {
    const confirmationAlertMessage = mount(
      <IntlProvider locale="en">
        <IntlConfirmationAlertMessage email="test@example.com" />
      </IntlProvider>,
    );

    const expectedMessage = 'Check Your Email'
                            + 'You entered test@example.com. If this email address is associated with your edX account, '
                            + 'we will send a message with password recovery instructions to this email address.'
                            + 'If you do not receive a password reset message after 1 minute, verify that you entered '
                            + 'the correct email address, or check your spam folder.'
                            + 'If you need further assistance, contact technical support.';

    expect(confirmationAlertMessage.find('#confirmation-alert').first().text()).toEqual(expectedMessage);
    expect(confirmationAlertMessage.find('#confirmation-alert').find('a').props().href).toEqual(supportLink);
  });
});
