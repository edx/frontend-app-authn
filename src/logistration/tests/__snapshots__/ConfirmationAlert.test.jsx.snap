// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationAlert should match default confirmation message snapshot 1`] = `
<div
  className="fade alert alert-success show"
  role="alert"
>
  <div
    className="text-success alert-heading h4"
  >
    <span>
      Check Your Email
    </span>
  </div>
  <span>
    You entered 
    <strong>
      test@example.com
    </strong>
    . If this email address is associated with your edX account, we will send a message with password recovery instructions to this email address. 
    <br />
    If you do not receive a password reset message after 1 minute, verify that you entered the correct email address, or check your spam folder.
    <br />
     If you need further assistance, 
    <a
      href="https://support.edx.org/hc/en-us/articles/206212088-What-if-I-did-not-receive-a-password-reset-message-"
      onClick={[Function]}
      target="_self"
    >
      <span>
        contact technical support
      </span>
    </a>
    .
  </span>
</div>
`;
